packages/hash/api/src/graphql/resolvers/task-executor/index.ts

```diff
@@ -172,45 +149,145 @@ export const executeGithubReadTask: ResolverFn<
     throw new ApolloError(
       "A task-executor wasn't started, so external tasks can't be started",
     );
-  } else {
-    const createdEntities = [];
-    try {
-      const airbyteRecords: AirbyteRecords = await taskExecutor.runTask(
-        Task.GithubRead,
-        config,
+  }
+
+  if (!user.shortname) {
```

Comment:
I thought the `loggedInAndSignedUpMiddleware` took care of ensuring these values exist. @benwerner01 is there an alternative way for us to supply the user context in the case of the middleware that knows the shortname exists when the user has completed signup?

Reply:
I think it does ensure they exist, but we could throw an assertion `!` in, which would then break if we ever changed the middleware (and honestly if we change the middleware are we going to remember to go through the full code path and look for such assertions), or we could do an additional cheap check here.

I've wanted us to rework the User type for a while now though, and establish something like `UninitializedUser`
https://app.asana.com/0/1201095311341924/1202944961125764/f

Reply:
`UninitializedUser` seems like a good way to handle this. I don't think we need to change anything here now